tests/integration_tests/functional/test_balloon.py

```diff
@@ -22,10 +22,22 @@
 MB_TO_PAGES = 256
 
 
-def get_rss_mem_by_pid(pid):
+def get_stable_rss_mem_by_pid(pid):
     """Get the RSS memory that a guest uses, given the pid of the guest."""
-    output = subprocess.check_output("pmap -X {}".format(pid), shell=True)
-    return int(output.decode('utf-8').split('\n')[-2].split()[1], 10)
+
+    def get_rss_from_pmap():
+        output = subprocess.check_output("pmap -X {}".format(pid), shell=True)
+        return int(output.decode('utf-8').split('\n')[-2].split()[1], 10)
+
+    rss = get_rss_from_pmap()
+    percentage_delta = 100
+    while percentage_delta > 0.5:
+        prev_rss = rss
+        rss = get_rss_from_pmap()
+        time.sleep(1)
+        percentage_delta = (abs(prev_rss - rss)/float(prev_rss)) * 100
```

Comment:
What if `percentage_delta` never drops below `0.5`?

`get_stable_rss_mem_by_pid` should have a couple of extra default parameters: `delta` and `timeout`: `def get_stable_rss_mem_by_pid(pid, delta=0.5, timeout=30)`

Reply:
Yes, added timeout and delta as parameters to the function. Also, I am raising an Exception in case of timeout. Let me know if that looks okay